### PR TITLE
Speeding up unitests

### PR DESCRIPTION
**Pull request by prabhuteja12**
Some of the unit-tests run for a long time. The slowest one is the `test_unique_classes`. This PR simplifies this, and marks a few other long running ones (mostly ones that download HF models/datasets) with a `pytest.mark.slow` decorator. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/renate/data/test_data_module.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def test_tiny_imagenet_data_module(tmpdir):
     assert train_data[0][0].size() == test_data[0][0].size() == (3, 64, 64)
 
 
+@pytest.mark.slow
```
Comment (wistuba) [inline test/renate/data/test_data_module.py:137]: why is this slow? is it the data download? can we use a smaller dataset? https://huggingface.co/datasets/ethos

File: test/renate/benchmark/test_experimentation_config.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
 from renate.models.prediction_strategies import ICaRLClassificationStrategy
 
 
+@pytest.mark.slow
```
Comment (wistuba) [inline test/renate/benchmark/test_experimentation_config.py:37]: how slow is this?

File: test/renate/utils/test_pytorch.py [hunk 3]
```diff
@@ -130,18 +129,12 @@ def test_complementary_indices(num_outputs, indices, expected_output):
...
-            scenario = ClassIncrementalScenario(
-                data_module=data_module, groupings=class_groupings, chunk_id=chunk_id
-            )
-            scenario.setup()
-            ds = scenario.val_data()
-            predicted_unique = unique_classes(ds)
-
-            assert predicted_unique == set(class_groupings[chunk_id])
+        for data_id in range(5):
+            module = DummyDataIncrementalDataModule(data_id, (10, 10), transform=None, val_size=0)
```
Comment (wistuba) [inline test/renate/utils/test_pytorch.py:133]: can we keep calling this `data_module`?

File: test/renate/utils/test_pytorch.py [hunk 4]
```diff
@@ -130,18 +129,12 @@ def test_complementary_indices(num_outputs, indices, expected_output):
...
+            module.setup()
```
Comment (wistuba) [inline test/renate/utils/test_pytorch.py:134]: I know `prepare_data` isn't doing anything, but this is how we call things. So I would like this line to stay.

File: test/renate/benchmark/test_experimentation_config.py (hunk 2 above)
Comment (prabhuteja12) [reply test/renate/benchmark/test_experimentation_config.py:37]: Instantiating any of the transformer variants requires a download of the model which is slow.

File: test/renate/data/test_data_module.py (hunk 1 above)
Comment (prabhuteja12) [reply test/renate/data/test_data_module.py:137]: Rotten tomatoes download is slow.

File: test/renate/utils/test_pytorch.py (hunk 4 above)
Comment (prabhuteja12) [reply test/renate/utils/test_pytorch.py:134]: Ok

File: test/renate/utils/test_pytorch.py (hunk 3 above)
Comment (prabhuteja12) [reply test/renate/utils/test_pytorch.py:133]: ok

**Review by wistuba — APPROVED** [review]

**Pull request merged**